### PR TITLE
Undo removing migration checks

### PR DESCRIPTION
**Pull request by Ank4n**
Re add the migration pre and post checks that was removed for PR https://github.com/paritytech/substrate/pull/12230. 

With @ruseinov's fix for sequential execution of migrate functions https://github.com/paritytech/substrate/pull/12319, we do not need to do this hack anymore.

### CODE REVIEW

File: frame/staking/src/migrations.rs [hunk 1]
```diff
@@ -129,6 +134,11 @@ pub mod v11 {
 
 		#[cfg(feature = "try-runtime")]
 		fn post_upgrade(_state: Vec<u8>) -> Result<(), &'static str> {
+			frame_support::ensure!(
+				StorageVersion::<T>::get() == crate::Releases::V11_0_0,
```
Comment (ruseinov) [inline frame/staking/src/migrations.rs:138]: I would compare that directly to V12

Comment (Ank4n) [reply frame/staking/src/migrations.rs:138]: Why so? This is the v11 migration. Your change should upgrade v11, execute `post_upgrade` and only then move on to the next migration right?

Comment (ggwpez) [reply frame/staking/src/migrations.rs:138]: I got confused as well; this diff changes two different migrations. It is correct.

Comment (ruseinov) [reply frame/staking/src/migrations.rs:138]: @ggwpez Ah I know what it is, this was disabled because try-runtime was  not handling things sequentially.

**Review by ruseinov — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]

**Comment by kianenigma** [comment]
just a minor note: https://github.com/paritytech/substrate/blob/master/docs/CONTRIBUTING.adoc#rules

**Comment by Ank4n** [comment]
> just a minor note: https://github.com/paritytech/substrate/blob/master/docs/CONTRIBUTING.adoc#rules

Noted

**Review by ggwpez — APPROVED** [review]

**Comment by Ank4n** [comment]
bot rebase

**Comment by Ank4n** [comment]
bot merge

**Pull request merged**